archive_upload/handlers/dsmc_handlers.py

```diff
@@ -545,26 +545,26 @@ class CreateDirHandler(BaseDsmcHandler):
     """
 
     @staticmethod
-    def _verify_unaligned(srcdir):
+    def _verify_required_dir(srcdir, required_path):
         """
-        Check that the archive contains the `Unaligned` symlink when running on biotanks.
-        The link should point to a proper directory.
+        Check that the required path exists as a directory or symlink to a proper directory.
 
         :param srcdir: The path to the archive which we should investigate
-        :return: True if `srcdir` contains a symlink `Unaligned` that points to a directory,
+        :param required_path: The path of the directory that is required
+        :return: True if `srcdir` contains the required paths,
                  False otherwise
         """
-        log.debug("Validating Unaligned...")
+        log.debug("Validating presence of {}...".format(required_path))
 
-        unaligned_link = os.path.join(srcdir, "Unaligned")
-        unaligned_dir = os.path.abspath(unaligned_link)
-
-        if not os.path.exists(unaligned_link) or not os.path.islink(unaligned_link):
+        path = os.path.join(srcdir, required_path)
+        if not os.path.exists(path):
             log.info(
-                "Expected link {} doesn't seem to exist or is broken. Aborting.".format(unaligned_link))
-            return False
-        elif not os.path.exists(unaligned_dir) or not os.path.isdir(unaligned_dir):
-            log.info("Expected directory {} doesn't seem to exist. Aborting.".format(unaligned_dir))
+                "Required path {} doesn't exist. Aborting.".format(path))
+
+        # In case of link make sure target dir exists.
```

Comment:
As far as I can see, this works the same way no matter if this is a link or the actual directory. If you agree I think we should change this comment.

Reply:
... I don't think `abspath` gets the path to the linked directory. Maybe that was intended?

Reply:
Fixed